docs/transforms.md

```diff
@@ -0,0 +1,114 @@
+## Transforms
+
+Transforms are operations applied on single records (i.e. multi-message transforms are not yet supported). 
+
+They are optional parts of a pipeline and, if present, they come in the form of processors (more information about 
+processor can be found in the [Architecture documentation](/docs/architecture.md)). As processors, transforms can be 
+attached either to a connector or to a pipeline. In other words, transforms can transform:
+1. Records coming out of a source connector (if the transform is attached to a source connector)
+2. Records coming into a destination connector (if the transform is attached to a destination connector)
+3. All records flowing through a pipeline (if the transform is attached to a pipeline)
+
+We'll go a little deeper into the case of a pipeline transform. At a high-level, source connectors, pipeline transforms
+and destination connector are laid out as in this image:
+
+![Pipeline](data/pipeline.svg)
+
+The main point here is that all pipeline transforms are "grouped together". Records from all sources will be sent to the
+transform in processor 1, then processor 2 etc. Pipeline transforms are chained in the order in which they were created.
+
+## Adding and configuring a transform
+
+Since a transform is a type of processor, all transforms are created through the `processors` endpoint. 
+Here's an example:
+
+```json lines
+POST /v1/processors
+{
+    // name of the transform
+    "name": "extractfieldpayload",
+    // type of processor: TYPE_TRANSFORM or TYPE_FILTER 
+    "type": "TYPE_TRANSFORM",
+    "parent": 
+    {
+        // type of parent: TYPE_CONNECTOR or TYPE_PIPELINE
+        "type": "TYPE_CONNECTOR",
+        // parent ID (connector ID in this case)
+        "id": "aed07589-44d8-4c68-968c-1f6c5197f13b" 
+    },
+    "config":
+    {
+        "settings":
+        {
+            // configuration map for this processor
+            "field": "name" 
+        }
+    }
+}
+```
+The request to create a processor is described in [api.proto](/proto/api/v1/api.proto).
```

Comment:
I'm thinking if we should rather link to the swagger doc, since we are referencing the HTTP API?

Reply:
Good one. I actually forgot we have them in the repo (I thought they are present only at build time and then included into a build.